v2/client/invoker.go

```diff
@@ -40,18 +42,21 @@ func (r *receiveInvoker) Invoke(ctx context.Context, m binding.Message, respFn p
 	var isFinished bool
 	defer func() {
 		if !isFinished {
-			if err2 := m.Finish(err); err2 == nil {
-				err = err2
+			if fErr := m.Finish(err); fErr != nil {
+				err = fErr
 			}
 		}
 	}()
 
-	e, err := binding.ToEvent(ctx, m)
-	if err != nil {
-		return err
-	}
+	var respMsg binding.Message
+	var result protocol.Result
+
+	e, eventErr := binding.ToEvent(ctx, m)
+	switch {
+	case eventErr != nil:
+		result = httpproto.NewResult(http.StatusBadRequest, "Failed to convert Message to Event: %v", eventErr)
```

Comment:
the client can't be coupled to http here, we can only test if it is an ack, nack, nor not nil.

Reply:
Are you suggesting that we can only return an error code without any context?

Reply:
Oh no, I think I got it now. I'll change that.